packages/bug-detectors/DEVELOPMENT.md

```diff
@@ -0,0 +1,75 @@
+# Bug Detector Development API
+
+Jazzer.js provides several tools for writing bug detectors.
+
+## Hooking Library Functions
+
+```typescript
+registerBeforeHook(target: string, pkg: string, async: boolean, hookFn: HookFn)
+registerReplaceHook(target: string, pkg: string, async: boolean, hookFn: HookFn)
+registerAfterHook(target: string, pkg: string, async: boolean, hookFn: HookFn)
+```
+
+## Adding Before/After Callbacks to the Fuzz Function
+
+```typescript
+registerAfterEachCallback(callback: () => void)
+registerBeforeEachCallback(callback: () => void)
+```
+
+These functions can be used to add callback functions that will always be
+executed before/after each fuzz test.
+
+## Adding Instrumentation Plugins
+
+```typescript
+registerInstrumentationPlugin(plugin: () => PluginTarget)
+```
+
+This function allows addition of instrumentation plugins to Jazzer.js. It
+expects a function that returns a `PluginTarget` from `"@babel/core"`. For an
+example of how to write an instrumentation plugin, see the
+[Prototype Pollution](internal/prototype-pollution.ts) bug detector and the
+[Jazzer.js instrumentation plugins](../instrumentor/plugins/).
+
+## Guiding the Fuzzing Process
+
+```typescript
+fuzzer.tracer.guideTowardsEquality(current: string, target: string, id: number)
+```
+
+Instructs the fuzzer to guide its mutations towards making `current` equal to
+`target`
+
+```typescript
+fuzzer.tracer.guideTowardsContainment(needle: string, haystack: string, id: number)
+```
+
+Instructs the fuzzer to guide its mutations towards making `haystack` contain
+`needle` as a substring.
+
+```typescript
+fuzzer.tracer.exploreState(state: number, id: number)
+```
+
+Instructs the fuzzer to attain as many possible values for the absolute value of
+`state` as possible.
+
+```typescript
+addDictionary(...libFuzzerDictionary: string[])
+```
+
+Whenever adding the above functions is not feasible, add values specific to your
+bug detector to a dictionary. The dictionary is used by the fuzzer as just
+another mutator, which means that the fuzzer will occasionally take values from
+the dictionary and replace parts of the whole input with it. This, using a
+dictionary is similar to the `guideTowardsContainment` function.
```

Comment:
Good point! My knowledge of libfuzzer is lacking. I am not sure if libfuzzer can figure out the exact offset at which it should put put the needle. So my assumption is that adding entries to the dictionary will eventually lead to those entries being contained in variables of interest! We should eventually benchmark this?
For now I can remove the link between dictionaries and the guidance functions, since I don't know the exact difference.